test/unit/connection/test-connection-ssl-max-tls1.0-reject.js

```diff
@@ -0,0 +1,32 @@
+var assert     = require('assert');
+var common     = require('../../common');
+var connection = common.createConnection({
+  port : common.fakeServerPort,
+  ssl  : {
+    ca         : common.getSSLConfig().ca,
+    maxVersion : 'TLSv1'
+  }
+});
+
+var server = common.createFakeServer({
+  ssl: {
+    minVersion: 'TLSv1.2'
+  }
+});
+
+server.listen(common.fakeServerPort, function(err) {
+  if (err) throw err;
+
+  connection.ping(function(err) {
+    var NODE_MAJOR_VERSION = process.versions.node.split('.')[0];
+    if (!err && NODE_MAJOR_VERSION >= 10) assert.fail('Expected to fail due to mismatched TLS versions for Node12+');
```

Comment:
I would guess that getting past this point (expecting it to fail) the test should assert that it is failing for the _reason_ we expect it to. Found quite a few bugs a while back when I added those kinds of tests in this lib.